src/qpoint.c

```diff
@@ -323,6 +333,122 @@ void qp_apply_annual_aberration(qp_memory_t *mem, double ctime, quat_t q, int in
+  quat_t q_step;
+
+  // deal with times
+  ctime2jd(ctime, jd_utc);
+
+#ifdef DEBUG
+  qp_print_memory(mem);
+#endif
+
+#ifdef DEBUG
+  printf("ctime %f, jd_utc %f %f\n", ctime, jd_utc[0], jd_utc[1]);
+#endif
+
+#ifdef DEBUG
+  qp_print_quat("state init", q);
+#endif
+
+  // apply "boresight rotation" (taking into account actual BS rotation)
+  qp_azelpsi_quat(az, el, psi, pitch, roll, q_step);
+  Quaternion_mul_left(q_step, q);
+#ifdef DEBUG
+  qp_print_quat("azel", q_step);
+  qp_print_quat("state azel", q);
+#endif
+
+  // apply refraction correction
+  // NB: per-detector refraction is not fully implemented!
+  // can only be done properly using the azel2radec functions
+  // otherwise, this is treated as a mean correction
+  qp_apply_refraction(mem, ctime, q, 0);
+
+  // apply diurnal aberration
+  // NB: same issue as refraction
+  // TODO propagate this to aberration step?
+  qp_apply_diurnal_aberration(mem, ctime, lat, q, 0);
+
+  // rotate to ITRS (by lon/lat)
+  if (qp_check_update(&mem->state_lonlat, ctime)) {
+    qp_lonlat_quat(lon, lat, mem->q_lonlat);
+#ifdef DEBUG
+    qp_print_quat("lonlat", mem->q_lonlat);
+#endif
+  }
+  if (qp_check_apply(&mem->state_lonlat)) {
+    Quaternion_mul_left(mem->q_lonlat, q);
+#ifdef DEBUG
+    qp_print_quat("state lonlat", q);
+#endif
+  }
+
+  // apply wobble correction (polar motion)
+  // or get dut1 from IERS bulletin
+  mjd_utc = jd2mjd(jd_utc[0]) + jd_utc[1];
+  if (qp_check_update(&mem->state_wobble, ctime)) {
+    qp_get_iers_bulletin_a(mem, mjd_utc, &mem->dut1, &x, &y);
+    ctime2jdtt(ctime, jd_tt);
+    qp_wobble_quat(jd_tt, x, y, mem->q_wobble);
+#ifdef DEBUG
+    qp_print_quat("wobble", mem->q_wobble);
+#endif
+  } else if (qp_check_update(&mem->state_dut1, ctime))
+    qp_get_iers_bulletin_a(mem, mjd_utc, &mem->dut1, &x, &y);
+  if (qp_check_apply(&mem->state_wobble)) {
+    Quaternion_mul_left(mem->q_wobble, q);
+#ifdef DEBUG
+    qp_print_quat("state wobble", q);
+#endif
+  }
+
+  // apply earth rotation
+  if (qp_check_update(&mem->state_erot, ctime)) {
+    // get ut1
+    jdutc2jdut1(jd_utc, mem->dut1, jd_ut1);
+    qp_erot_quat(jd_ut1, mem->q_erot);
+#ifdef DEBUG
+    qp_print_quat("erot", mem->q_erot);
+#endif
+  }
+  if (qp_check_apply(&mem->state_erot)) {
+    Quaternion_mul_left(mem->q_erot, q);
+#ifdef DEBUG
+    qp_print_quat("state erot", q);
+#endif
+  }
+
+  // apply nutation/precession/frame bias correction
+  if (qp_check_update(&mem->state_npb, ctime)) {
+    if (jd_tt[0] == 0) ctime2jdtt(ctime, jd_tt);
+    qp_npb_quat(jd_tt, mem->q_npb, mem->accuracy);
+#ifdef DEBUG
+    qp_print_quat("npb", mem->q_npb);
+#endif
+  }
+  if (qp_check_apply(&mem->state_npb)) {
+    Quaternion_mul_left(mem->q_npb, q);
+#ifdef DEBUG
+    qp_print_quat("state npb", q);
+#endif
+  }
+
+  // apply annual aberration
+  // ~20 arcsec max
+  if (mem->mean_aber)
+    qp_apply_annual_aberration(mem, ctime, q, 0);
+
+#ifdef DEBUG
+  qp_print_quat("state final", q);
+#endif
+}
+
```

Comment:
To avoid code duplication with `qp_azel2quat`, I'd have `qp_azel2quat` call `qp_azelpsi2quat` with the `psi` argument set to 0.

Reply:
Do you want `qp_azel2quat`  to call `qp_azelpsi2quat`, or should `qp_azel2bore` directly call `qp_azelpsi2quat` with `psi=0`, and remove `qp_azel2quat` entirely? Is there any chance there's someone out there calling `qp_azel2quat` directly, whose code we'd break?

Reply:
There are six other calls to `qp_azel2quat` in the code that would have to be modified to `qp_azelpsi2quat(psi=0)`, so the changes are a bit deeper, but possibly results in cleaner code? Let me know what you decide and I can implement.

Reply:
I would leave `qp_azel2quat` in place for backward compatibility, just in case, but use `qp_azelpsi2quat(psi=0)` everywhere internal as you suggest.

Reply:
Now that I'm looking at this code, we should probably also have `azelpsi2*` variants of all of the `azel2*` functions for completeness...